source/write/transaction.txt

```diff
@@ -0,0 +1,168 @@
+
+Create a ``Session`` by using the ``MongoDB\Client::startSession()``
+method on your ``Client`` instance. The {+php-library+} provides a
+Convenient Transaction API to manage the transaction lifecyle. Use the
+``MongoDB\with_transaction()`` method to run custom callback within a
+transaction. The ``with_transaction()`` method starts the transaction,
+then either commits it or ends it if there are errors. The
+:ref:`php-txn-example` section of this guide demonstrates how to use
+this API to perform a transaction. 
+
+Alternatively, you can have more control over your transaction lifecyle
+by using the methods provided by the ``Session`` class. The
+following table describes these methods:
+
+.. list-table::
+   :widths: 25 75
+   :stub-columns: 1
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction on this session. The session
+         must be passed into each operation within the transaction, or
+         the operation will run outside of the transaction.
+       |
+       | You can set transaction options by passing an options parameter.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. This method returns an
+         error if there is no active transaction for the session or the
+         transaction was previously ended.
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. This method returns an
+         error if there is no active transaction for the session or if the
+         transaction was committed or ended.
+
+.. _php-txn-example:
+
+Transaction Example
+-------------------
+
+The following code defines the ``transferTxn()`` callback function that
+modifies data in the ``saving_accounts``, ``checking_accounts``, and ``receipts``
+collections of the ``bank`` database:
+
+.. literalinclude:: /includes/write/transaction.php
+   :language: php
+   :dedent:
+   :start-after: begin-callback
+   :end-before: end-callback
+
+Then, run the following code to perform the transaction. This code
+completes the following actions:
+
+1. Creates a session from the client by using the ``startSession()`` method.
+#. Calls the ``with_transaction()`` method to manage the transaction,
+   passing the session and the callback as parameters.
+
+.. io-code-block::
+
+   .. input:: /includes/write/transaction.php
+      :language: php
+      :dedent:
+      :start-after: begin-txn
+      :end-before: end-txn
+
+   .. output:: 
+      :language: console
+      :visible: false
+
+      Successfully performed transaction!
+
+Additional Information
+----------------------
+
+To learn more about the concepts mentioned in this guide, see the
+following pages in the {+mdb-server+} manual:
+
+- :manual:`Transactions </core/transactions/>`
+- :manual:`Server Sessions </reference/server-sessions/>`
+- :manual:`Read Isolation, Consistency, and Recency
+  </core/read-isolation-consistency-recency/#causal-consistency>`
```

Comment:
Q: did you intentionally link to the Casual Consistency section of this page? I think you can just link to the main page
```suggestion
- :manual:`Read Isolation, Consistency, and Recency
  </core/read-isolation-consistency-recency/>`
```